cmpv2/src/certified_key_pair.rs

```diff
@@ -0,0 +1,99 @@
+//! Certificate key pair-related types
+
+use alloc::vec::Vec;
+use der::{Choice, Sequence};
+
+use crmf::controls::{EncryptedValue, PkiPublicationInfo};
+
+use crate::header::CmpCertificate;
+use crate::status::PkiStatusInfo;
+
+/// The `CertifiedKeyPair` type is defined in [RFC 4210 Section 5.3.4]
+///
+/// ```text
+///  CertifiedKeyPair ::= SEQUENCE {
+///      certOrEncCert       CertOrEncCert,
+///      privateKey      [0] EncryptedValue      OPTIONAL,
+///      publicationInfo [1] PKIPublicationInfo  OPTIONAL }
+/// ```
+///
+/// [RFC 4210 Section 5.3.4]: https://www.rfc-editor.org/rfc/rfc4210#section-5.3.4
+#[derive(Clone, Debug, Eq, PartialEq, Sequence)]
+#[allow(missing_docs)]
+pub struct CertifiedKeyPair {
+    pub cert_or_enc_cert: CertOrEncCert,
+    #[asn1(
+        context_specific = "0",
+        tag_mode = "EXPLICIT",
+        constructed = "true",
+        optional = "true"
+    )]
+    pub priv_key: Option<EncryptedValue>,
+    #[asn1(
+        context_specific = "1",
+        tag_mode = "EXPLICIT",
+        constructed = "true",
+        optional = "true"
+    )]
+    pub publication_info: Option<PkiPublicationInfo>,
+}
+
+/// The `CertOrEncCert` type is defined in [RFC 4210 Section 5.3.4]
+///
+/// ```text
+///  CertOrEncCert ::= CHOICE {
+///      certificate     [0] CMPCertificate,
+///      encryptedCert   [1] EncryptedValue }
+/// ```
+///
+/// [RFC 4210 Section 5.3.4]: https://www.rfc-editor.org/rfc/rfc4210#section-5.3.4
+#[derive(Clone, Debug, Eq, PartialEq, Choice)]
+#[allow(missing_docs)]
+#[allow(clippy::large_enum_variant)]
```

Comment:
Might be worth boxing the large enum variant

Reply:
Fixed for this struct. There are a few more. Will do those too with similar approach.